src/Tasks/Common/Logger.cs

```diff
@@ -0,0 +1,123 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Diagnostics;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+
+namespace Microsoft.NET.Build.Tasks
+{
+    /// <summary>
+    /// Replacement and abstraction for <see cref="TaskLoggingHelper"/> in our
+    /// build tasks.
+    /// </summary>
+    /// <remarks>
+    /// Source compatible with ususal Log.LogXxx MSBuild task code. (Subset of
+    /// API chosen based on actual usage in SDK, and with a deliberate goal of
+    /// eliminating some of the excessive overloading in TaskLoggingHelper.
+    ///
+    /// <see cref="Message"/> replaces the need for overloads taking over 10
+    /// arguments.
+    ///
+    /// Also, string[] is used instead of object[] to avoid issues like passing
+    /// the importance out of order as a format argument.
+    ///
+    /// <see cref="Log"/> allows choosing Error/Warning/Message dynamically at a
+    /// single call site.
+    ///
+    /// Extracts error codes from the message prefix, and enforces that all of
+    /// our messages have a NETSDK code.
+    ///
+    /// Example:
+    ///   C#
+    ///     Log.LogError(Strings.SomethingIsWrong);
+    ///
+    ///   Strings.resx:
+    ///     | SomethingIsWrong | NETSDK1234: Something is wrong. |
```

Comment:
The pipes didn't make it clear to me that this wasn't just one long string.  Perhaps something like this would be clearer:

```
  Strings.resx:
    Resource Name: SomethingIsWrong
    Resource Value: NETSDK1234: Something is wrong.
```